### DISCUSSION TITLE
Turn Global FX-to-preset links to Unit-wide FX-to-preset links

### DISCUSSION THREAD

**Discussion by tompad2** [helgoboss/helgobox#1144] [Q&A] [answered]
Have a bunch of Global FX-to-preset links I want to turn to Unit-wide FX-to-preset links.
Is there an easy way to do this? ...not re-doing the presets...

**Comment by helgoboss** [comment, marked as answer]
Even if you would recreate the FX-to-preset links manually, you would not have to re-create the (linked-to) presets as well. But I guess you know that already.

There is a way to move the global FX-to-preset links into one specific instance, but it involves some text file editing.

1. Open the global FX-to-preset link JSON file in a proper text editor (not Wordpad please ;): `REAPER_RESOURCE_PATH/Data/helgoboss/realearn/auto-load-configs/fx.json`. Should look a bit like this:
  ```json5
  {
    "links": [
      // some links
    ]
  }
  ```
2. Open the ReaLearn unit into which you want to import your global links.
3. Press "Export to clipboard => Export unit as JSON"
4. Paste the result into another text editor window. Should look a bit like this:
  ```json5
  {
    "kind": "Unit",
    "version": "2.16.3",
    "value": {
      "version": "2.16.3",
      "id": "6tNAfa8i",
      "name": "Main",
      "stayActiveWhenProjectInBackground": "OnlyIfBackgroundProjectIsRunning",
      "sendFeedbackOnlyIfArmed": true,
      "defaultGroup": {},
      "defaultControllerGroup": {},
      "instancePresetLinkConfig": {
        "links": [
          {
            "fx": {
              "name": "VST: ReaLimit (Cockos)"
            },
            "presetId": "bla/foo"
          }
        ]
      },
      "instanceFx": {
        "address": "Focused"
      }
    }
  }
  ```
5. Replace the "links" array in the export (step 4) with the link array from the global file (step 1)
6. Copy the result to the clipboard
7. Press "Import from clipboard" in the ReaLearn unit

That should be it.

**Reply by tompad2** [reply]
Excellent!
Thanks Benjamin - you are a rock!

I love textfile-editing (when I understand what I am doing)! ;-)

**Reply by tompad2** [reply]
Thanks!
No I didn't know... learning new things with ReaLearn :-)

**Comment by tompad2** [comment]
Your great explaining made me wonder another thing, maybe with same sort of solution.

I tested your DAW control preset and I found out it have some nice features:

1. Coloring the LCD after track color
2. When I select one track it become the first track on controller with the other following.

I have done my own DAW control preset for a long time ago and it have some things I want/need
to have.

My question is, is it possible to merge the 2 DAW control presets to one with the goodies from
both of the presets?? I have looked in my own preset but it was a long time ago I made it so
I dont really understand what have been done ;-)

**Reply by helgoboss** [reply]
That can only be done manually. There's no way ReaLearn can know which mappings you want to keep/remove/modify.
